api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,278 @@
+    // Note: This is only temporary replacement for appendable string
+    std::string hex_string = "00-";
+    for (int i = 0; i < 32; i++)
+    {
+      hex_string += trace_id[i];
+    }
+    hex_string += "-";
+    for (int i = 0; i < 16; i++)
+    {
+      hex_string += span_id[i];
+    }
+    hex_string += "-";
+    for (int i = 0; i < 2; i++)
+    {
+      hex_string += trace_flags[i];
+    }
+    setter(carrier, kTraceParent, hex_string);
+  }
+
+  static void InjectImpl(Setter setter, T &carrier, const SpanContext &span_context)
+  {
+    InjectTraceParent(span_context, carrier, setter);
+  }
+
+  static bool IsValidHex(nostd::string_view string_view)
+  {
+    for (int i = 0; i < string_view.length(); i++)
+    {
+      if (!(string_view[i] >= '0' && string_view[i] <= '9') &&
+          !(string_view[i] >= 'a' && string_view[i] <= 'f'))
+        return false;
+    }
+    return true;
+  }
+
+  static SpanContext ExtractContextFromTraceParent(nostd::string_view trace_parent)
+  {
+    if (trace_parent.length() != kHeaderSize || trace_parent[kVersionBytes] != '-' ||
+        trace_parent[kVersionBytes + kTraceIdBytes + 1] != '-' ||
+        trace_parent[kVersionBytes + kTraceIdBytes + kSpanIdBytes + 2] != '-')
+    {
+      std::cout << "Unparseable trace_parent header. Returning INVALID span context." << std::endl;
+      return SpanContext(false, false);
+    }
+    nostd::string_view version = trace_parent.substr(0, kHeaderElementLengths[0]);
+    nostd::string_view trace_id =
+        trace_parent.substr(kHeaderElementLengths[0] + 1, kHeaderElementLengths[1]);
+    nostd::string_view span_id = trace_parent.substr(
+        kHeaderElementLengths[0] + kHeaderElementLengths[1] + 2, kHeaderElementLengths[2]);
+    nostd::string_view trace_flags = trace_parent.substr(
+        kHeaderElementLengths[0] + kHeaderElementLengths[1] + kHeaderElementLengths[2] + 3);
+
+    if (version == "ff")
+    {
+      return SpanContext(false, false);
+    }
+
+    if (trace_id == "00000000000000000000000000000000" || span_id == "0000000000000000")
+    {
+      return SpanContext(false, false);
+    }
+
+    // validate ids
+    if (!IsValidHex(version) || !IsValidHex(trace_id) || !IsValidHex(span_id) ||
+        !IsValidHex(trace_flags))
+      return SpanContext(false, false);
+
+    TraceId trace_id_obj       = GenerateTraceIdFromString(trace_id);
+    SpanId span_id_obj         = GenerateSpanIdFromString(span_id);
+    TraceFlags trace_flags_obj = GenerateTraceFlagsFromString(trace_flags);
+    return SpanContext(trace_id_obj, span_id_obj, trace_flags_obj, true);
+  }
+
+  static SpanContext ExtractImpl(Getter getter, const T &carrier)
+  {
+    nostd::string_view trace_parent = getter(carrier, kTraceParent);
+    if (trace_parent == "")
+    {
+      return SpanContext(false, false);
+    }
+    SpanContext context_from_parent_header = ExtractContextFromTraceParent(trace_parent);
+    return context_from_parent_header;
```

Comment:
You can just `return ExtractContextFromTraceParent(...);`

Reply:
Changed